kubernetes/client/api/apps_v1_api.py

```diff
@@ -3887,7 +3887,7 @@ def patch_namespaced_deployment_with_http_info(self, name, namespace, body, **kw
 
         # HTTP header `Content-Type`
         header_params['Content-Type'] = self.api_client.select_header_content_type(  # noqa: E501
-            ['application/json-patch+json', 'application/merge-patch+json', 'application/strategic-merge-patch+json'])  # noqa: E501
+            ['application/json-patch+json', 'application/merge-patch+json', 'application/strategic-merge-patch+json', 'application/apply-patch+yaml'])  # noqa: E501
```

Comment:
Hi. I have `K8S 1.16`. I noticed one interesting bug. If I call `patch_namespaced_deployment` through the client to `DELETE` 1 or more environment variables, then the request is successful, however, the deployment itself is not updated.

After a little research, I found out that the `select_header_content_type` method called here returns just the 1st element of this array. I also found out that if you put the 2nd element of an array in the header, then the deployment is updated perfectly.

```suggestion
            ['application/merge-patch+json', 'application/json-patch+json', 'application/strategic-merge-patch+json', 'application/apply-patch+yaml'])  # noqa: E501
```